src/main/java/sorald/sonar/RuleViolation.java

```diff
@@ -1,28 +1,35 @@
 package sorald.sonar;
 
 import java.util.Objects;
-import org.sonar.java.AnalyzerMessage;
 
-/** Facade around {@link org.sonar.java.AnalyzerMessage} */
-public class RuleViolation {
-    private final AnalyzerMessage message;
-
-    RuleViolation(AnalyzerMessage message) {
-        this.message = message;
-    }
+/** Representation of a violation of some Sonar rule */
+public abstract class RuleViolation {
 
     /** @return The line number related to the rule violation. */
-    public int getLineNumber() {
-        return Objects.requireNonNull(message.getLine());
-    }
+    public abstract int getLineNumber();
 
     /** @return The name of the file that was analyzed. */
-    public String getFileName() {
-        return message.getInputComponent().key().replace(":", "");
-    }
+    public abstract String getFileName();
 
     /** @return The name of the check class that generated this warning. */
-    public String getCheckName() {
-        return message.getCheck().getClass().getSimpleName();
+    public abstract String getCheckName();
+
+    /** @return The key of the rule that is violated. */
+    public abstract String getRuleKey();
```

Comment:
@khaes-kth The miner will most likely need this method. Perhaps I should make a separate PR with this functionality.